accio-base/src/main/java/io/accio/base/metadata/BasicFunctions.java

```diff
@@ -0,0 +1,19 @@
+package io.accio.base.metadata;
+
+import com.google.common.collect.ImmutableList;
+
+import static io.accio.base.metadata.Function.Argument.argument;
+import static io.accio.base.metadata.Function.builder;
+import static io.accio.base.type.TimestampType.TIMESTAMP;
+import static io.accio.base.type.VarcharType.VARCHAR;
+
+public class BasicFunctions
+{
+    private BasicFunctions() {}
+
+    public static final Function DATE_TRUNC = builder()
+            .setName("date_trunc")
+            .setArguments(ImmutableList.of(argument("field", VARCHAR), argument("source", TIMESTAMP)))
+            .setReturnType(TIMESTAMP)
```

Comment:
Let's change back to return `DATE` type. It's an important usage for BigQuery user. We will fix `date_trunc` type issue in another PR.

Reply:
Related issue #468